File: services/validator-api/Models/VerificationModels.cs
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace ValidatorApi.Models;

public enum Verdict
{
    Verified,
    Suspicious,
    Inconclusive
}

public class MissingSpan
{
    public int StartElapsedMs { get; set; }
    public int EndElapsedMs { get; set; }
    public string Reason { get; set; } = string.Empty;
}

public class VerificationResult
{
    public Verdict Verdict { get; set; }
    public string SessionId { get; set; } = string.Empty;
    public int Threshold { get; set; }
    public int ToleranceMs { get; set; }
    public int IntervalMs { get; set; }
    public int ExpectedSamples { get; set; }
    public int MatchedSamples { get; set; }
    public double MatchRatio { get; set; }
    public double AvgDistance { get; set; }
    public int MaxDistance { get; set; }
    public List<MissingSpan> MissingSpans { get; set; } = new();
    public List<string> Notes { get; set; } = new();
    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
    public VerificationDebugMetrics? Debug { get; set; }
}

public class VerificationDebugMetrics
{
    public string SessionId { get; set; } = string.Empty;
    public int SessionSamplingIntervalMs { get; set; }
    public int ToleranceMs { get; set; }
    public int Threshold { get; set; }
    public int ReferenceHashCount { get; set; }
    public int ExtractedFrameCount { get; set; }
    public DebugElapsedMsRange? ExtractedElapsedMsRange { get; set; }
    public List<MatcherWindowStat> MatcherWindowStats { get; set; } = new();
}

public class DebugElapsedMsRange
{
    public int MinElapsedMs { get; set; }
    public int MaxElapsedMs { get; set; }
}

public class MatcherWindowStat
{
    public int RefElapsedMs { get; set; }
    public int CandidateCountInWindow { get; set; }
    public int? BestMinDistance { get; set; }
}

File: apps/validator-portal/src/app/models/verification.models.ts
export type Verdict = 'Verified' | 'Suspicious' | 'Inconclusive';

export interface MissingSpan {
  startElapsedMs: number;
  endElapsedMs: number;
  reason: string;
}

export interface VerificationResult {
  verdict: Verdict;
  sessionId: string;
  threshold: number;
  toleranceMs: number;
  intervalMs: number;
  expectedSamples: number;
  matchedSamples: number;
  matchRatio: number;
  avgDistance: number;
  maxDistance: number;
  missingSpans: MissingSpan[];
  notes: string[];
  debug?: VerificationDebugMetrics;
}

export interface VerificationDebugMetrics {
  sessionId: string;
  sessionSamplingIntervalMs: number;
  toleranceMs: number;
  threshold: number;
  referenceHashCount: number;
  extractedFrameCount: number;
  extractedElapsedMsRange?: DebugElapsedMsRange;
  matcherWindowStats: MatcherWindowStat[];
}

export interface DebugElapsedMsRange {
  minElapsedMs: number;
  maxElapsedMs: number;
}

export interface MatcherWindowStat {
  refElapsedMs: number;
  candidateCountInWindow: number;
  bestMinDistance: number | null;
}
